1808 return a default image when there is no diagram info in bpmn model

constantin-ciobotaru opened the pull request:
- move diagram generator logic in a dedicated service ProcessDiagramGeneratorWrapper
- ask if there is graphic info in bpmn model before generating it
- add spring configuration for context with process engine
- add tests for ProcessDiagramGeneratorWrapper
- add the possibility to specify custom fonts for generator to use
- return a default image when there is no digram info in bpmn model
- add tests for the default image

Salaboy commented on activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/test/resources/test-process-diagram.properties:
```diff
@@ -0,0 +1,2 @@
+activiti.engine.diagram.activity.font=Lucida
```
@constantin-ciobotaru Are these the default values? I would like to avoid users need to set up this thing.

Salaboy commented on activiti-cloud-services-runtime-bundle/activiti-cloud-services-core/src/main/resources/image/na.svg:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
+<!-- Generator: Adobe Illustrator 18.1.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
```
@constantin-ciobotaru this might cause problems. It might be good to find a Creative Commons licensed SVG.

constantin-ciobotaru replied:
Ups, I didn't realize it :(
Thanks.

constantin-ciobotaru commented:
I'm not sure it's a good idea to move it in engine. I don't see it as an engine functionality. Maybe it would be better just to avoid the engine to throw the NPE, but to throw a dedicated exception, and let the caller to decide which would be the best handling of this exception. What do you think? Anyway I'll take a look in engine to see if there is any code there already that is based on the current functionality.

Salaboy commented:
@constantin-ciobotaru the key point is that I'm not suggesting to move it to the engine, but to the library in charge of dealing with the image generation, which is a separate jar. So people using directly that API should have a similar experience that they ones calling the rest layer

constantin-ciobotaru commented:
Changes added by [c698836](https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/51/commits/c698836be32865a65e2aab791ab4d213378e1306) :
1. moved the default digram image to activiti-image-generator
2. moved hasGraphicInfo to BpmnModel and rename it to hasDiagramInterchangeInfo
3. by default throw ActivitiInterchangeInfoNotFoundException when hasDiagramInterchangeInfo is false
4. added properties to allow default diagram image to be returned and to provide a custom image for that
5. return 204 when ActivitiInterchangeInfoNotFoundException is thrown
